zerver/lib/url_decoding.py

```diff
@@ -23,4 +23,4 @@ def is_same_server_message_link(url: str) -> bool:
     section = fragment_parts[1]
     ends_with_near_message_id = fragment_parts[-2] == "near" and fragment_parts[-1].isdigit()
 
-    return category == "narrow" and section in {"stream", "dm"} and ends_with_near_message_id
+    return category == "narrow" and section in {"channel", "dm"} and ends_with_near_message_id
```

Comment:
Hmm, this function might be used in a way that we should be supporting the legacy values too, in that drafts may still have the old syntax. Maybe I can just add a bonus commit for that detail.

Reply:
Thinking about this a bit more, I think it's likely to be sufficiently rare that this actually happens that it's not worth introducing temporary legacy code around it, given the impact will be just a bit less nice desktop/push notification alert.